pkg/subctl/cmd/gather/resource.go

```diff
@@ -98,3 +99,13 @@ func gatherConfigMaps(info Info, namespace string, listOptions metav1.ListOption
 		Resource: "configmaps",
 	}, namespace, listOptions)
 }
+
+func scrubSensitiveData(info Info, dataString string) string {
+	if !info.IncludeSecrets {
+		dataString = strings.Replace(dataString, info.Submariner.Spec.BrokerK8sApiServer, "##redeacted-api-server##", -1)
```

Comment:
`info.Submariner` will be nil if not installed, in which case we need to get it from `ServiceDiscovery`. It would be nice to refactor the common bits into a separate resource that is always installed. In lieu of that, I'll send a patch after this to try to handle this smoothly.

Reply:
I was going to add a patch with ServiceDiscovery, plan to use that for some of lighthouse stuff and can then reuse it if Submariner is not present. Can refactor this code once that comes in.

Reply:
I pushed #1283 - I can refactor this with that one.